examples/asr/speech_to_text.py

```diff
@@ -49,7 +51,7 @@ def main(args):
     model_config['AudioToTextDataLayer_eval']['manifest_filepath'] = args.eval_dataset
     asr_model.setup_training_data(model_config['AudioToTextDataLayer'])
     asr_model.setup_validation_data(model_config['AudioToTextDataLayer_eval'])
-    asr_model.setup_optimization(optim_params={'lr': 0.0003})
+    asr_model.setup_optimization(optim_params={'optimizer': args.optimizer, 'lr': args.lr, 'opt_args': args.opt_args})
```

Comment:
I suppose this is out of scope of PR, but these three lines look holly out of line with pytorch lightning code. Just do all of this in init(). I fail to see the reason we need to do this separately.

Reply:
@blisc are you proposing to have models.__init__() take: (1) model hyper parameters, (2) optimizer hyper parameters and (3) train/test/eval data parameters instead of having setup_* functions?

Reply:
@blisc I came to exactly the same conclusions yesterday - thus my email.

I think the solution is to properly _parametrize_ NeMo Models.

Reply:
We no longer need to manually extract the kwargs from the parsed args, `vars(args)` is concise and serves the same purpose.